<launch>
    <node pkg="lidar_ohyejun" type="gigacha_lidar_clustering" name="gigacha_lidar_clustering" output="screen">
        
        <param name="sensor_height" value="1.55"/>
        <param name="verbose" value="false"/>

        <param name="num_iter" value="5"/>
        <param name="num_lpr" value="20"/>
        <param name="num_min_pts" value="10"/>
        <param name="th_seeds" value="0.2"/>
        <param name="th_dist" value="0.20"/>
        <param name="max_r" value="80.0"/>
        <param name="min_r" value="2.7"/>
        <param name="uprightness_thr" value="0.707"/>
        <param name="adaptive_seed_selection_margin" value="-1.1"/>
        
        <param name="czm/num_zones" value="4"/>
        
        <rosparam param="czm/num_sectors_each_zone">[16, 32, 54, 32]</rosparam>
        
        <rosparam param="czm/mum_rings_each_zone">[2, 4, 4, 4]</rosparam>
        
        <rosparam param="czm/elevation_thresholds">[0.523, 0.994, 1.085, 1.15]</rosparam>
        <rosparam param="czm/flatness_thresholds">[0.0, 0.0012, 0.0088, 0.0]</rosparam>
        
    </node>
</launch>